torchvision/utils.py

```diff
@@ -382,6 +389,132 @@ def draw_keypoints(
+
+    u = flow_uv[0, :, :]
+    v = flow_uv[1, :, :]
+    rad = torch.sqrt(torch.square(u) + torch.square(v))
+    rad_max = torch.max(rad)
+    epslion = 1e-5
+    u = u / (rad_max + epslion)
+    v = v / (rad_max + epslion)
+    return _flow_uv_to_colors(u, v)
+
+
+@torch.no_grad()
+def _flow_uv_to_colors(
+    u: torch.Tensor,
+    v: torch.Tensor,
+) -> torch.Tensor:
+
+    """
+    Applies the flow color wheel to (possibly clipped) flow components u and v.
+
+    Args:
+        u (torch.Tensor): Input horizontal flow of shape [H, W]
+        v (torch.Tensor): Input vertical flow of shape [H, W]
+
+    Returns:
+       img (Tensor[3, H, W]): Flow visualization image.
+    """
+
+    flow_image = torch.zeros((3, u.shape[0], u.shape[1]), dtype=torch.uint8)
+    colorwheel = _make_colorwheel()
+    # shape [55x3]
+    ncols = colorwheel.shape[0]
+    rad = torch.sqrt(torch.square(u) + torch.square(v))
+
+    a = torch.atan2(-v, -u) / torch.pi
+    fk = (a + 1) / 2 * (ncols - 1)
+    k0 = torch.floor(fk).to(torch.long)
+    k1 = k0 + 1
+    k1[k1 == ncols] = 0
+    f = fk - k0
+
+    for i in range(colorwheel.shape[1]):
+        tmp = colorwheel[:, i]
+        col0 = tmp[k0] / 255.0
+        col1 = tmp[k1] / 255.0
+        col = (1 - f) * col0 + f * col1
+        idx = rad <= 1
+        col[idx] = 1 - rad[idx] * (1 - col[idx])
+        col[~idx] = col[~idx] * 0.75  # out of range
+        ch_idx = i
+        flow_image[ch_idx, :, :] = torch.floor(255 * col)
+    return flow_image
+
+
+def _make_colorwheel() -> torch.Tensor:
+    """
+    Generates a color wheel for optical flow visualization as presented in:
+    Baker et al. "A Database and Evaluation Methodology for Optical Flow" (ICCV, 2007)
+    URL: http://vision.middlebury.edu/flow/flowEval-iccv07.pdf.
+
+    Returns:
+        colorwheel (Tensor[255, 3]): Colorwheel Tensor.
+    """
+
+    RY = 15
+    YG = 6
+    GC = 4
+    CB = 11
+    BM = 13
+    MR = 6
+
+    ncols = RY + YG + GC + CB + BM + MR
+    colorwheel = torch.zeros((ncols, 3))
+    col = 0
+
+    # RY
+    colorwheel[0:RY, 0] = 255
+    colorwheel[0:RY, 1] = torch.floor(255 * torch.arange(0, RY) / RY)
+    col = col + RY
+    # YG
+    colorwheel[col : col + YG, 0] = 255 - torch.floor(255 * torch.arange(0, YG) / YG)
+    colorwheel[col : col + YG, 1] = 255
+    col = col + YG
+    # GC
+    colorwheel[col : col + GC, 1] = 255
+    colorwheel[col : col + GC, 2] = torch.floor(255 * torch.arange(0, GC) / GC)
+    col = col + GC
+    # CB
+    colorwheel[col : col + CB, 1] = 255 - torch.floor(255 * torch.arange(CB) / CB)
+    colorwheel[col : col + CB, 2] = 255
+    col = col + CB
+    # BM
+    colorwheel[col : col + BM, 2] = 255
+    colorwheel[col : col + BM, 0] = torch.floor(255 * torch.arange(0, BM) / BM)
+    col = col + BM
+    # MR
+    colorwheel[col : col + MR, 2] = 255 - torch.floor(255 * torch.arange(MR) / MR)
+    colorwheel[col : col + MR, 0] = 255
+    return colorwheel
+
+
```

Comment:
I believe this file is getting a bit big. I don't know what's the convention and limits for number of lines in a file.
Maybe we should consider making `utils` a folder a expose accordingly.

This will allow to define colormaps like these in separate file.